docker-build

```diff
@@ -1,32 +1,41 @@
 #!/bin/sh
+# shellcheck shell=dash
 
 set -e
 
-if [ ! -z "$CIRCLECI" ]; then
-  REPO="$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME"
-  BRANCH="$CIRCLE_BRANCH"
-  SHA="$CIRCLE_SHA1"
-else
-  >&2 echo "This script only works on CircleCI right now."
-  exit 1
-fi
-
-if [[ -n "$NO_EXTRA_TAGS" ]]
-then
-    extra_tags=false
-else
-    extra_tags=true
-fi
-
 # Usage prints the help for this command.
 usage() {
   >&2 echo "Usage:"
-  >&2 echo "    docker-build command"
+  >&2 echo "    docker-build [OPTIONS]"
+  >&2 echo "    docker-build COMMAND [OPTIONS]"
+  >&2 echo ""
+  >&2 echo "Docker CLI wrapper that injects defaults for a more consistent"
+  >&2 echo "CI process. Currently only works in CircleCI."
+  >&2 echo ""
+  >&2 echo "Running docker-build with no COMMAND will do build-with-cache"
+  >&2 echo "and then push automatically"
+  >&2 echo ""
+  >&2 echo "Any OPTIONS specified will be passed to 'docker build."
   >&2 echo ""
   >&2 echo "Commands:"
-  >&2 echo "    pull   Attempt to pull the latest cache tag to build a layer cache"
-  >&2 echo "    build  Build a docker image from this repo and tag it with the git sha and branch"
-  >&2 echo "    push   Push the built image to the docker registry."
+  >&2 echo "    pull              Attempt to pull the latest cache tag to build a layer cache"
+  >&2 echo "    build             Build a docker image from this repo and tag it with the git sha and branch"
+  >&2 echo "    build-with-cache  Build with automatic --cache-from"
+  >&2 echo "    push              Push the built image to the docker registry."
+  >&2 echo ""
+  >&2 echo "Examples:"
+  >&2 echo "    Basic usage:"
+  >&2 echo "      $ docker-build"
+  >&2 echo ""
+  >&2 echo "    Build, but don't tag with branch and SHA1 tags:"
+  >&2 echo "      $ NO_EXTRA_TAGS=1 docker-build build"
+  >&2 echo ""
+  >&2 echo "    Build and push an image with a multi-stage Dockerfile:"
+  >&2 echo "      $ docker-build --target main"
+  >&2 echo ""
+  >&2 echo "    Build with buildkit (Docker >= 19.03):"
+  >&2 echo "      $ DOCKER_BUILDKIT=1 docker-build"
+  >&2 echo ""
```

Comment:
Just a style question, but any reason you didn't use a HEREDOC string? Something like

```sh
usage() {
  >&2 echo <<EOF
Usage:
    docker-build command
etc.
```

Reply:
I was just rolling with what was already here, I agree a HEREDOC would be way cleaner.